### DISCUSSION TITLE
Issue in creating exe : Module Not Found in PyInstaller-Built NiceGUI

### DISCUSSION THREAD

**Discussion by 07rohanraj** [zauberzeug/nicegui#4141] [Q&A]
Hi,

I’m encountering a ModuleNotFoundError for the frontend.home module when running my PyInstaller-built NiceGUI application. Here's a detailed overview of my setup and the issue:

**![Screenshot 2024-12-24 154415](https://github.com/user-attachments/assets/c6b2776f-2ea9-4f65-8169-53435b3afcd7)**

**Project Structure:**

    SBOM-Generator-Tool/
    ├── assets/
    ├── backend/
    │   ├── cyclonedx-cli/
    │   └── SBOM_Generator_GUI.py
    ├── frontend/
    │   ├── __init__.py
    │   ├── home/
    │   │   ├── __init__.py
    │   │   └── some_file.py
    │   └── other modules...
    ├── _res/
    └── other directories...

**Command used for pyInstaller:**

    PyInstaller --onefile --windowed --clean --additional-hooks-dir=hooks --hidden-import=numpy --hidden-import=pandas --collect-all nicegui --add-data "assets/*;assets" --add-data "_res/*;cyclonedx/schema/_res" --add-data "frontend/*;frontend" .\backend\SBOM_Generator_GUI.py

**Error:**
    After building the .exe file, when I run it, I get the following error:
    
    Traceback (most recent call last):
      File "SBOM_Generator_GUI.py", line 73, in <module>
      File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
      File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
      File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
      File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
    ModuleNotFoundError: No module named 'frontend.home'

**Backend code has these imports:**

      from frontend.home.home_ui import Home_ui
      from frontend.create_sbom.create_sbom_ui import create_sbom
      from frontend.Add_edit.add_edit_ui import add_edit
      from frontend.loading_ui.loading_cpe_ui import loading_cpe_ui

**What I've Tried:**

  - Explicitly added frontend using --add-data "frontend/*;frontend"
  - Used --debug=imports but couldn't find any missing imports from frontend

**Questions:**
    - Is there a specific way NiceGUI handles folder structures or imports that I might be missing?
    - How do I ensure that all modules in frontend are included correctly in the PyInstaller bundle?
    - Could this be related to NiceGUI's handling of paths or PyInstaller's packaging?

**Comment by ed2050** [comment]
`frontend.home` appears to be your code.  Can't tell if the problem is there or with nicegui.

Have you tried the [pyinstaller docs](https://pyinstaller.org/en/stable/)?  They have several pages on common problems and troubleshooting.  I also see a number of [forum posts](https://github.com/orgs/pyinstaller/discussions) on "no module found" errors.
